presto-client/src/main/java/com/facebook/presto/client/StatementClient.java

```diff
@@ -251,7 +253,7 @@ public boolean advance()
                 throw requestFailedException("fetching next", request, response);
             }
         }
-        while ((System.nanoTime() - start) < MINUTES.toNanos(2) && !isClosed());
+        while ((System.nanoTime() - start) < clientRequestTimeout.getValue(NANOSECONDS) && !isClosed());
```

Comment:
One minor change is that returns a `double`, but it is cleaner to do the comparison using a `long`, so we do the conversion once in the constructor (which fails up front if the conversion would overflow):
```java
this.requestTimeoutNanos = session.getClientRequestTimeout().roundTo(NANOSECONDS);